js/modules/k6/http/response.go

```diff
@@ -141,3 +144,77 @@ func (res *HTTPResponse) Html(selector ...string) html.Selection {
 	}
 	return sel
 }
+
+func (res *HTTPResponse) SubmitForm(args ...goja.Value) (*HTTPResponse, error) {
+	rt := common.GetRuntime(res.ctx)
+
+	formSelector := "form"
+	submitSelector := "[type=\"submit\"]"
+	var fields map[string]goja.Value
+	var requestParams goja.Value
+	if len(args) > 0 {
+		params := args[0].ToObject(rt)
+		for _, k := range params.Keys() {
+			switch k {
+			case "formSelector":
+				formSelector = params.Get(k).String()
+			case "submitSelector":
+				submitSelector = params.Get(k).String()
+			case "fields":
+				if rt.ExportTo(params.Get(k), &fields) != nil {
+					fields = nil
+				}
+			case "params":
+				requestParams = params.Get(k)
+			}
+		}
+	}
+
+	form := res.Html(formSelector)
+	if form.Size() == 0 {
+		common.Throw(rt, fmt.Errorf("no form found for selector '%s' in response '%s'", formSelector, res.URL))
+	}
+
+	methodAttr := form.Attr("method")
+	var requestMethod string
+	if methodAttr == goja.Undefined() {
+		// Use GET by default
+		requestMethod = HTTP_METHOD_GET
+	} else {
+		requestMethod = strings.ToUpper(methodAttr.String())
+	}
+
+	actionAttr := form.Attr("action")
+	var requestUrl goja.Value
+	if actionAttr == goja.Undefined() {
+		// Use the url of the response if no action is set
+		requestUrl = rt.ToValue(res.URL)
+	} else {
+		// Resolve the action url from the response url
+		responseUrl, _ := url.Parse(res.URL)
```

Comment:
Please add error checking to the two `url.Parse()` calls and call `common.Throw(...)` if there's an error.